### PR TITLE
[Core][Process] adding Check

### PR DESCRIPTION
**Pull request by philbucher**
I discussed with @jcotela and we though it would make sense to have a `Check`-fct in the Processes

@adityaghantasala this could be used for checking the MPCs when you apply / create them in a process

=> requires #3001

### CODE REVIEW

File: kratos/processes/process.h
Comment (adityaghantasala) [inline kratos/processes/process.h:129]: why not `virtual bool Check()` ?

Comment (philbucher) [reply kratos/processes/process.h:129]: other classes return ints
e.g. ModelPart, SolvingStrategy

File: kratos/processes/structured_mesh_generator_process.cpp
Comment (RiccardoRossi) [inline kratos/processes/structured_mesh_generator_process.cpp]: let's call this Check as it is called everywhere else...

File: kratos/processes/structured_mesh_generator_process.h
Comment (RiccardoRossi) [inline kratos/processes/structured_mesh_generator_process.h]: let's call this Check()

**Review by RiccardoRossi — COMMENTED** [review]
in principle we (@KratosMultiphysics/technical-committee )  agree, however we would call this Check for uniformity with other portions of the code

File: kratos/processes/structured_mesh_generator_process.cpp
Comment (philbucher) [reply kratos/processes/structured_mesh_generator_process.cpp]: In this case it works a bit differently from what I could see, also it is private, therefore I did not want to change the behavior
@pooyan-dadvand you wrote this, maybe you can comment on it

File: kratos/python_scripts/analysis_stage.py
Comment (philbucher) [inline kratos/python_scripts/analysis_stage.py:162]: > in principle we (@KratosMultiphysics/technical-committee ) agree, however we would call this Check for uniformity with other portions of the code

this is exactly what I am proposing to do here

File: kratos/processes/structured_mesh_generator_process.cpp
Comment (RiccardoRossi) [reply kratos/processes/structured_mesh_generator_process.cpp]: Oh, is specific of this class... ok

Comment (pooyan-dadvand) [reply kratos/processes/structured_mesh_generator_process.cpp]: I put it private to maintain the public interface of Process intact. Now you can make it public without any problem.

**Comment by philbucher** [comment]
ping @KratosMultiphysics/technical-committee

**Review by pooyan-dadvand — DISMISSED** [review]
:+1:

**Comment by philbucher** [comment]
thanks @pooyan-dadvand 
This PR also contains #3001 which is not approved yet
I am not sure now if I should merge them separately or all at once...?

**Comment by pooyan-dadvand** [comment]
Yep, #3001 is still in queue... sorry for that....

I would separate it if is not a large effort. If not I would not merge it until it approves....

**Comment by philbucher** [comment]
conflicts solved and #3001 is merged
@pooyan-dadvand can you reapprove pls?

**Review by pooyan-dadvand — APPROVED** [review]
👍

**Comment by philbucher** [comment]
Thanks

**Pull request merged**
